### PR TITLE
Add FaultConfiguration class

### LINKED ISSUE #133: Add `FaultConfiguration` class
## User stories
As a developer of the fault injector I want a basic `FaultConfigurator` class to let the fault type specific configurator classes inherit from it.

## Acceptance Criteria
- [x] add `FaultConfigurator` class
- [x] add fields to `FaultConfigurator` class
- [ ] add class to table-list in constants.py

### PR DESCRIPTION
**Pull request by instant-sky**
Fixes #133 

Description (what might a PO or reviewer want to know?)

## PR checklist

- [x] Acceptance criteria fulfilled
- [x] ~Additional features are tested~
- [x] Docs (code, wiki & diagrams) updated
- [x] ~Breaking changes anounced~
- [x] Dev-branch has been merged into local branch to resolve conflicts
- [x] Tests and linter have passed AFTER local merge
- [x] Another dev reviewed and approved

### CODE REVIEW

**Review by Lietze — APPROVED** [review]
Good job, every common attribute from the wiki is implemeted now.

**Review by antonykamp — APPROVED** [review]

File: src/fault_injector/fault_types/fault.py [hunk 1]
```diff
@@ -37,3 +40,18 @@ def next_tick(self, tick: int):
             self.inject_fault(self.component)
         elif tick == self.end_tick:
             self.resolve_fault(self.component)
+
+
+class FaultConfiguration(BaseModel):
```
Comment (antonykamp) [inline src/fault_injector/fault_types/fault.py]: Missing function for serialization and deserialization 😇

**Review by antonykamp — CHANGES_REQUESTED** [review]

**Review by antonykamp — COMMENTED** [review]
The class is looking pretty good now 🥳 
You wanna add tests? Or do you think its not necessary for this config? 🙃

File: src/fault_injector/fault_types/fault.py [hunk 2]
```diff
@@ -37,3 +40,21 @@ def next_tick(self, tick: int):
...
```
Comment (antonykamp) [inline src/fault_injector/fault_types/fault.py]: Please add the FaultConfiguration class to the array of db-tables in `src/constants.py` 😇

Comment (antonykamp) [reply src/fault_injector/fault_types/fault.py]: Not needed, because FaultConfiguration should be abstract and not available as table.

**Review by antonykamp — APPROVED** [review]
Looking f*cking good now 🥳

**Pull request merged**